### DISCUSSION TITLE
Chord keybindings

### DISCUSSION THREAD

**Discussion by darrenburns** [Textualize/textual#4596] [Ideas]
I'd love to be able to create chorded keybindings. I think this would solve much of the namespacing issue we have with limited keybings available.

Maybe I could define a binding like this:

```python
Binding("ctrl+k/u", ...)
```

(That syntax probably isn't the best - but it hopefully gets the idea across)

This would mean if the user presses `ctrl+k` followed by `u`, the action would fire.

Alternatively:

```python
BINDINGS = [
    Binding("x", ...),
    Chord("ctrl+k", [
        Binding("a", ...),
        Binding("b", ...),
    ])
]
```

In this example, `ctrl+k` followed by `a` would trigger an action. `ctrl+k` followed by `b` would trigger another action.

This would integrate into the footer, so after pressing `ctrl+k`, the footer would update to only show `a` and `b`.

Sometimes these systems have things like timeouts e.g. `Chord("ctrl+k", timeout=5, bindings=[...])` would mean after pressing `ctrl+k`, the bindings inside the chord would be available for 5 seconds. Perhaps pressing escape could short circuit it and take you back to the parent bindings.

**Comment by 1dancook** [comment]
The tree-like syntax in your second example looks great! It also looks very easy to continue nesting.

One thing to implement is a keypress to "back out" of the chord. Usually this is ESC.
